annex/chapter1/adv_examples/src/example7.cpp

```diff
@@ -0,0 +1,62 @@
+/****************************************************************************
+*    1 event loop
+*    Listener to print the content of the event and the thread id
+*    Publish and subscriber for other types of data: int, my own strct, etc.
+*    Number as the output
+*****************************************************************************/
+
+#include <iostream>
+#include <klepsydra/high_performance/event_loop_middleware_provider.h>
+#include <thread>
+#include <chrono>
+# include "int_data_producer.h"
+#include "position.h"
+#include "position_publisher_service.h"
+#include "position_subscriber_service.h"
+
+
+int main() {
+    kpsr::high_performance::EventLoopMiddlewareProvider<16> eventloop(nullptr);
+
+    eventloop.start();
+    
+    eventloop.getSubscriber<int>("temperature receiver")->registerListener("listener", [](const int & message) {
+        std::cout <<"temperature received t : " << message << std::endl;
+        std::thread::id this_id = std::this_thread::get_id();
+        std::cout << "temperaturelistener id: " << this_id << '\n';
+    });
+
+    PositionPublisherService positionPublisherService(eventloop.getPublisher<Position>("positionCheck", 0, nullptr, nullptr));
+    PositionSubscriberService positionSubscriberService(eventloop.getSubscriber<Position>("positionCheck"));
+    
+/*    eventloop.getSubscriber<Position>("position receiver")->registerListener("listener", [](const Position & message) {
```

Comment:
Remove commented code that is not being used.

Reply:
Done